wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift

```diff
@@ -337,20 +278,76 @@ extension ConversationCreationController: AddParticipantsConversationCreationDel
             values.participants = users
 
         case .create:
-            var allParticipants = values.participants
-            allParticipants.insert(selfUser)
+            guard let userSession = ZMUserSession.shared() else { return }
+            let service = ConversationService(context: userSession.viewContext)
 
-            delegate?.conversationCreationController(
-                self,
-                didSelectName: values.name,
-                participants: values.participants,
+            let users = values.participants
+                .union([selfUser])
+                .materialize(in: userSession.viewContext)
+
+            service.createGroupConversation(
+                name: values.name,
+                users: Set(users),
                 allowGuests: values.allowGuests,
                 allowServices: values.allowServices,
                 enableReceipts: values.enableReceipts,
-                encryptionProtocol: values.encryptionProtocol
-            )
+                messageProtocol: values.encryptionProtocol == .proteus ? .proteus : .mls
+            ) { [weak self] in
+                guard let self = self else { return }
+
+                switch $0 {
+                case .success(let conversation):
+                    delegate?.conversationCreationController(
+                        self,
+                        didCreateConversation: conversation
+                    )
+
+                case .failure(.networkError(.nonFederatingDomains(let domains))):
```

Comment:
This error isn't listed in swagger so I didn't add it. Reverse engineering our code suggests that this error is possible when creating new conversation, so I will add it just in case.

Reply:
Actually, it is but it's a 403 not 412. But indeed we're not showing the alert so I'll add that.

Reply:
done